Espanso crashes under Hyprland v0.32.3 and above

**Describe the bug**
Espanso crashes as soon as the systemd service starts. Running `espanso daemon` does output an error (see below).

**To Reproduce**
Steps to reproduce the behavior:
1. Update Hyprland from `v0.30.0` to `v0.32.3`, `v0.33.0`, or `v0.33.1` and restart.

**Expected behavior**
FWIW Hyprland `v0.30.0` runs espanso perfectly.

**Logs**
```
17:57:29 [daemon(10402)] [INFO] reading configs from: "/home/bk/.config/espanso"
17:57:29 [daemon(10402)] [INFO] reading packages from: "/home/bk/.config/espanso/match/packages"
17:57:29 [daemon(10402)] [INFO] using runtime dir: "/home/bk/.cache/espanso"
17:57:29 [daemon(10402)] [INFO] system info: Arch Linux v - kernel: 6.6.4-AMD-znver2
17:57:29 [daemon(10402)] [WARN] keyboard layout watcher couldn't determine active layout.
17:57:29 [daemon(10402)] [INFO] watching for changes in path: "/home/bk/.config/espanso"
17:57:29 [daemon(10402)] [INFO] espanso version: 2.1.8
17:57:29 [daemon(10402)] [INFO] spawning the worker process...
17:57:29 [daemon(10402)] [INFO] binded to IPC unix socket: /home/bk/.cache/espanso/espansodaemonv2.sock
17:57:29 [worker(10411)] [INFO] reading configs from: "/home/bk/.config/espanso"
17:57:29 [worker(10411)] [INFO] reading packages from: "/home/bk/.config/espanso/match/packages"
17:57:29 [worker(10411)] [INFO] using runtime dir: "/home/bk/.cache/espanso"
17:57:29 [worker(10411)] [INFO] system info: Arch Linux v - kernel: 6.6.4-AMD-znver2
17:57:29 [worker(10411)] [INFO] binded to IPC unix socket: /home/bk/.cache/espanso/espansoworkerv2.sock
17:57:29 [worker(10411)] [INFO] using WaylandAppInfoProvider
17:57:29 [worker(10411)] [INFO] monitoring the status of the daemon process
17:57:29 [worker(10411)] [INFO] detection module will use this keyboard layout: [R=, M=, L=us, V=, O=]
17:57:29 [worker(10411)] [INFO] using EVDEVSource
17:57:30 [worker(10411)] [INFO] Querying modifier status...
17:57:30 [worker(10411)] [ERROR] thread 'detect thread' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 0, kind: Uncategorized, message: "Success" }': espanso-detect/src/evdev/sync/wayland.rs:193
17:57:30 [worker(10411)] [ERROR] thread 'engine thread' panicked at 'unable to receive from the init_rx channel: RecvError': espanso/src/cli/worker/engine/funnel/mod.rs:124
17:57:30 [worker(10411)] [ERROR] Unable to block the LinuxEventLoop: receiving on an empty and disconnected channel
17:57:30 [worker(10411)] [ERROR] thread 'main' panicked at 'unable to run main eventloop: receiving on an empty and disconnected channel': espanso/src/cli/worker/mod.rs:160
17:57:30 [daemon(10402)] [ERROR] received unexpected exit code from worker 101, exiting
```

**Setup information**

Arch Linux

`uname -a`:

```
6.6.4-AMD-znver2 #1 SMP PREEMPT_DYNAMIC Sun Dec  3 08:15:20 CET 2023 x86_64 GNU/Linux
```

`lspci -v`:

```
0b:00.0 VGA compatible controller: Advanced Micro Devices, Inc. [AMD/ATI] Ellesmere [Radeon RX 470/480/570/570X/580/580X/590] (rev ef) (prog-if 00 [VGA controller])
    Subsystem: Tul Corporation / PowerColor Radeon RX 570 4G [Red Dragon]
    Flags: bus master, fast devsel, latency 0, IRQ 164, IOMMU group 27
    Memory at d0000000 (64-bit, prefetchable) [size=256M]
    Memory at e0000000 (64-bit, prefetchable) [size=2M]
    I/O ports at f000 [size=256]
    Memory at fce00000 (32-bit, non-prefetchable) [size=256K]
    Expansion ROM at fce40000 [disabled] [size=128K]
    Capabilities: <access denied>
    Kernel driver in use: amdgpu
    Kernel modules: amdgpu
```

`espanso --version`:

```
espanso 2.1.8
```

espanso config:

```
keyboard_layout: { layout: us }
key_delay: 0
search_shortcut: off
search_trigger: off
show_notifications: false
```

More info here:

  - https://github.com/hyprwm/Hyprland/issues/3858
  - https://github.com/hyprwm/Hyprland/issues/4085

--- a/espanso-detect/Cargo.toml
+++ b/espanso-detect/Cargo.toml
@@ -24,7 +24,7 @@ widestring = "0.4.3"
 [target.'cfg(target_os="linux")'.dependencies]
 libc = "0.2.85"
 scopeguard = "1.1.0"
-sctk = { package = "smithay-client-toolkit", version = "0.15.4", optional = true }
+sctk = { package = "smithay-client-toolkit", version = "0.16.1", optional = true }
 
 [build-dependencies]
 cc = "1.0.73"

--- a/espanso-detect/src/evdev/sync/wayland.rs
+++ b/espanso-detect/src/evdev/sync/wayland.rs
@@ -67,10 +67,7 @@ pub fn get_modifiers_state() -> Result<Option<super::ModifiersState>> {
    * Keyboard initialization
    */
 
-  let mut seats = Vec::<(
-    String,
-    Option<(wl_keyboard::WlKeyboard, calloop::RegistrationToken)>,
-  )>::new();
+  let mut seats = Vec::<(String, Option<wl_keyboard::WlKeyboard>)>::new();
 
   // first process already existing seats
   for seat in env.get_all_seats() {
@@ -89,8 +86,8 @@ pub fn get_modifiers_state() -> Result<Option<super::ModifiersState>> {
           RepeatKind::System,
           move |event, _, _| keyboard_event_handler(event, &result_clone),
         ) {
-          Ok((kbd, repeat_source)) => {
-            seats.push((name, Some((kbd, repeat_source))));
+          Ok(kbd) => {
+            seats.push((name, Some(kbd)));
           }
           Err(e) => {
             error!("Failed to map keyboard on seat {} : {:?}.", name, e);
@@ -127,8 +124,8 @@ pub fn get_modifiers_state() -> Result<Option<super::ModifiersState>> {
           RepeatKind::System,
           move |event, _, _| keyboard_event_handler(event, &result_clone),
         ) {
-          Ok((kbd, repeat_source)) => {
-            *opt_kbd = Some((kbd, repeat_source));
+          Ok(kbd) => {
+            *opt_kbd = Some(kbd);
           }
           Err(e) => {
             eprintln!(
@@ -138,10 +135,9 @@ pub fn get_modifiers_state() -> Result<Option<super::ModifiersState>> {
           }
         }
       }
-    } else if let Some((kbd, source)) = opt_kbd.take() {
+    } else if let Some(kbd) = opt_kbd.take() {
       // the keyboard has been removed, cleanup
       kbd.release();
-      loop_handle.remove(source);
     }
   });
 

